### PR TITLE
improve exception handling & documentation

### PR DESCRIPTION
**Pull request by individual-it**
1. document in more details which exceptions are thrown
2. throw better exceptions in some cases

### CODE REVIEW

File: src/Ocis.php [hunk 1]
```diff
@@ -182,6 +191,11 @@ public function listAllDrives(
      * Get all drives that the current user is a regular member of
      *
      * @return array<Drive>
+     *
+     * @throws BadRequestException
+     * @throws ForbiddenException
+     * @throws NotFoundException
+     * @throws UnauthorizedException
      * @throws \Exception
```
Comment (phil-davis) [inline src/Ocis.php:199]: Do we still need this general Exception line?

File: src/Drive.php [hunk 2]
```diff
@@ -277,6 +294,10 @@ public function uploadFile(string $path, string $content): bool
      * @param resource|string|null $resource file resource pointing to the file to be uploaded
      *
      * @return bool
+     * @throws BadRequestException
+     * @throws ForbiddenException
+     * @throws NotFoundException
+     * @throws UnauthorizedException
      * @throws \Exception
```
Comment (phil-davis) [inline src/Drive.php:301]: Do we still need lines like this?
Are there other "unusual" exceptions that are known to be possible?

File: src/Ocis.php [hunk 3]
```diff
@@ -248,8 +267,10 @@ public function getDriveById(string $driveId): Drive
      * @param int $quota in bytes
      * @return Drive
      * @throws \Exception
```
Comment (phil-davis) [inline src/Ocis.php:269]: Maybe this line can be deleted?

**Review by phil-davis — APPROVED** [review]

File: src/Drive.php (hunk 2 above)
Comment (individual-it) [reply src/Drive.php:301]: we throw the \Exception ourself in this case in line 308. I could change it to `InvalidParameterException`

**Comment by individual-it** [comment]
@phil-davis I've refactored the code, so that it never throws general Exceptions

**Review by phil-davis — APPROVED** [review]

**Pull request merged**